test/e2e/all_in_one_test.go

```diff
@@ -119,28 +84,50 @@ func allInOneWithIngressTest(t *testing.T, f *framework.Framework, ctx *framewor
 		return err
 	}
 
-	ingress, err := WaitForIngress(t, f.KubeClient, namespace, "my-jaeger-with-ingress-query", retryInterval, timeout)
-	if err != nil {
-		return err
-	}
+	var url string
+	var httpClient http.Client
+	if isOpenShift(t) {
+		err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, name, 1, retryInterval, 3 * timeout)
+		if err != nil {
+			t.Errorf("Error waiting for deployment of %s: %v\n", name, err)
+			return err
+		}
 
-	if len(ingress.Status.LoadBalancer.Ingress) != 1 {
-		return fmt.Errorf("Wrong number of ingresses. Expected 1, was %v", len(ingress.Status.LoadBalancer.Ingress))
+		routeUrl, err := findRouteUrl(t, f, name)
+		if err != nil {
+			return err
+		}
+
+		url = fmt.Sprintf("https://%s/api/services", routeUrl)
+		transport := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		httpClient= http.Client{Timeout: 30 * time.Second, Transport: transport}
+	} else {
+		ingress, err := WaitForIngress(t, f.KubeClient, namespace, "my-jaeger-with-ingress-query", retryInterval, timeout)
+		if err != nil {
+			return err
+		}
+
+		if len(ingress.Status.LoadBalancer.Ingress) != 1 {
```

Comment:
Should this test be performed in the openshift branch aswell?

Reply:
@objectiser what would be the equivalent for a route?  len(route.Status.Ingress)?

Reply:
Think so.